app/services/weights_and_measures.rb

```diff
@@ -26,9 +26,15 @@ def system
   def self.variant_unit_options
     available_units_sorted.flat_map do |measurement, measurement_info|
       measurement_info.filter_map do |scale, unit_info|
+        # Our code is based upon English based number formatting
+        # Some language locales like +hu+ uses a comma(,) for decimal separator
+        # While in English, decimal separator is represented by a period.
+        # e.g. en: 0.001, hu: 0,001
+        # Hence the results become "weight_0,001" for hu while or code recognizes "weight_0.001"
         scale_clean =
           ActiveSupport::NumberHelper.number_to_rounded(scale, precision: nil, significant: false,
-                                                               strip_insignificant_zeros: true)
+                                                               strip_insignificant_zeros: true,
+                                                               locale: :en)
         [
           "#{I18n.t(measurement)} (#{unit_info['name']})", # Label (eg "Weight (g)")
           "#{measurement}_#{scale_clean}", # Scale ID (eg "weight_1")
```

Comment:
I'm not sure exactly, we can tweak the keys of the `UNITS` hash such that we don't need to use the helper to round off-scale values here. i.e. already round off keys here. But I'm reluctant to do so because it's being used in other places and not sure if it's a good idea to do it this quickly.
https://github.com/chahmedejaz/openfoodnetwork/blob/a50be52cded204db8ba67e6dee4bdb4888d2d4a1/app/services/weights_and_measures.rb#L65-L86

The current solution works just like we would have expected in the old UI. So, maybe we can go with it for now. 🤷